File: src/me/ch2/bmi.py
height_cm = float(input("身長は?(cm) "))
weight = float(input("体重は?(kg) "))

# bmi
height = height_cm / 100
bmi = weight / (height * height)
print(f"BMIは{bmi:.2f}です")

if bmi < 18.5:
    print("痩せ型")
elif bmi < 25:
    print("普通")
elif bmi < 30:
    print("肥満(軽)")
elif bmi < 35:
    print("肥満(中)")
elif bmi < 40:
    print("肥満(重)")
else:
    print("肥満(極)")

File: src/me/ch2/bmi.rs
fn main() {
    let height_cm = input("Height (cm): ");
    let weight_kg = input("Weight (kg): ");

    // bmi
    let height = height_cm / 100.0;
    let bmi = weight_kg / (height * height);
    println!("BMI: {}", bmi);

    // bmi category
    if bmi < 18.5 {
        println!("Underweight");
    } else if bmi < 25.0 {
        println!("Normal");
    } else if bmi < 30.0 {
        println!("Obese Class I");
    } else if bmi < 35.0 {
        println!("Obese Class II");
    } else if bmi < 40.0 {
        println!("Obese Class III");
    } else {
        println!("Obese Class IV");
    }
}

fn input(prompt: &str) -> f64 {
    println!("{}", prompt);
    let mut s = String::new();
    std::io::stdin()
        .read_line(&mut s)
        .expect("Failed to read line");
    return s.trim().parse().expect("Failed to parse f64");
}
